tests/e2e/testsLibrary/WorkspaceHandlingTests.ts

```diff
@@ -80,30 +79,33 @@ export class WorkspaceHandlingTests {
 
     public obtainWorkspaceNameFromStartingPage(): void {
         test('Obtain workspace name from workspace loader page', async() => {
-            try {
-                Logger.info('Waiting for workspace name on workspace loader page');
-                await this.driverHelper.waitVisibility(WorkspaceHandlingTests.READY_TO_READ_WORKSPACE_NAME_LOCATOR, TimeoutConstants.TS_WAIT_LOADER_PRESENCE_TIMEOUT);
-
-                const timeout: number = TimeoutConstants.TS_IDE_LOAD_TIMEOUT;
-                const polling: number = TestConstants.TS_SELENIUM_DEFAULT_POLLING;
-                const attempts: number = Math.ceil(timeout / polling);
-                let startingWorkspaceLineContent: string;
-
-                for (let i: number = 0; i < attempts; i++) {
-                    startingWorkspaceLineContent = await this.driverHelper.getDriver().findElement(WorkspaceHandlingTests.START_WORKSPACE_PAGE_NAME_LOCATOR).getAttribute('innerHTML');
-
-                    // cutting away leading text
-                    WorkspaceHandlingTests.workspaceName = startingWorkspaceLineContent.substring('Starting workspace '.length).trim();
-                    if (WorkspaceHandlingTests.workspaceName !== '') {
-                        Logger.info(`Obtained workspace name from workspace loader page: ${WorkspaceHandlingTests.workspaceName}`);
-                        break;
+            const timeout: number = TimeoutConstants.TS_IDE_LOAD_TIMEOUT;
+            const polling: number = TestConstants.TS_SELENIUM_DEFAULT_POLLING;
+            const attempts: number = Math.ceil(timeout / polling);
+            let startingWorkspaceLineContent: string = '';
+
+            for (let i: number = 0; i < attempts; i++) {
+                try {
+                    startingWorkspaceLineContent = await this.driverHelper.getDriver().findElement(WorkspaceHandlingTests.WORKSPACE_NAME_LOCATOR).getText();
+                } catch (err) {
+                    if (err instanceof error.StaleElementReferenceError) {
```

Comment:
is it a good solution to skip StaleElementReferenceError? I think we can get this error in a case of some bug. Maybe it will be better to give more time to the driver to wait this webelement?

Reply:
Current logic polls for the element every second for 20 seconds

Reply:
Nevermind, `error.TimeoutError` doesn't get thrown with this method. Current logic should be sufficient.

If you agree with the approach, @nallikaea please resolve the threads.

Reply:
okay